template/alpha_formula.py

```diff
@@ -22,28 +22,33 @@ def __init__(self, _startdate, data, parameters):
         self.k_days = parameters['k']
         self.mid_last_k_days = {
             instrument: deque(
-                (datum.buy + datum.sell) / 2
-                for datum in price_data[-self.k_days - 1: -1])
+                ((datum.buy + datum.sell) / 2
+                 for datum in price_data[-self.k_days - 1: -1]
+                 if all((datum.buy, datum.sell))))
```

Comment:
may there are too many parentheses here?

Reply:
Even it seems too many parentheses, it really need that much QQ.

Reply:
I guess it is ok to delete the pair out of
```
(datum.buy + datum.sell) / 2
    for datum in price_data[-self.k_days - 1: -1]
    if all((datum.buy, datum.sell))
```

Reply:
@ioumu8 Can you check that?

Reply:
Yes! this can work.